xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2156,6 +2156,7 @@ void CVideoPlayer::CheckContinuity(CCurrentStream& current, DemuxPacket* pPacket
        fabs(this_dts - that_dts) < DVD_MSEC_TO_TIME(1000))
     {
       m_offset_pts += correction;
+      m_VideoPlayerVideo->SetOffsetPts(m_offset_pts);
```

Comment:
I tested with removing the if (pts > 0) condition and this also fixes the issue. If you agree that this is a better solution, I can update this commit to that effect?